spring-rabbit/src/main/java/org/springframework/amqp/rabbit/support/micrometer/RabbitTemplateObservation.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.amqp.rabbit.support.micrometer;
+
+import io.micrometer.common.docs.KeyName;
+import io.micrometer.observation.docs.DocumentedObservation;
+
+/**
+ * Spring RabbitMQ Observation for {@link org.springframework.amqp.rabbit.core.RabbitTemplate}.
+ *
+ * @author Gary Russell
+ * @since 3.0
+ *
+ */
+public enum RabbitTemplateObservation implements DocumentedObservation {
+
+	/**
+	 * {@link org.springframework.kafka.core.KafkaTemplate} observation.
+	 */
+	TEMPLATE_OBSERVATION {
+
+		@Override
+		public String getName() {
+			return "spring.rabbit.template";
```

Comment:
You can use the `getDefaultConvention` method - https://github.com/spring-cloud/spring-cloud-task/blob/main/spring-cloud-task-core/src/main/java/org/springframework/cloud/task/configuration/observation/TaskDocumentedObservation.java#L32 which gives you better customization options

Reply:
Hold on. Then it means that my question about `RabbitTemplateObservationConvention` as a custom convention from end-user remains opened.
According that task feature we should have an extra interface and have it default impl for use-cases like this and contextual name.
The interface would be used as a point of customization from end-user, not a class as it is right now.

Reply:
Yeah I modified my response here https://github.com/spring-projects/spring-amqp/pull/1500#discussion_r966029706